src/ray/common/memory_monitor.cc

```diff
@@ -0,0 +1,112 @@
+// Copyright 2022 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <tuple>
+#include "ray/common/memory_monitor.h"
+#include <fstream>      // std::ifstream
+#include "ray/util/logging.h"
+#include "ray/common/ray_config.h"
+
+
+namespace ray {
+
+MemoryMonitor::MemoryMonitor(float capacity_threshold,
+    uint64_t monitor_interval_ms,
+    MemoryUsageRefreshCallback monitor_callback)
+  : usage_threshold_(capacity_threshold),
+    monitor_callback_(monitor_callback),
```

Comment:
Can you handle when callback == nullptr? (either require RAY_CHECK(monitor_callback_ != nullptr) or add if?)

Reply:
sg